batch/batch/cloud/gcp/worker/disk.py

```diff
@@ -11,22 +11,29 @@
 
 
 class GCPDisk(CloudDisk):
-    def __init__(self, name: str, zone: str, project: str, instance_name: str, size_in_gb: int, mount_path: str):
+    def __init__(
+        self,
+        name: str,
+        zone: str,
+        project: str,
+        instance_name: str,
+        size_in_gb: int,
+        mount_path: str,
+        compute_client: aiogoogle.GoogleComputeClient,  # BORROWED
```

Comment:
Can you explain the changes you made for borrowing the compute client for creating the attached GCPDisks? I thought we did that for a reason. I can try and go back through the git history to find it.

Reply:
I wasn't aware that there was a specific reason to create a new `ComputeClient` per disk. The reason I made this change was so that we wouldn't have O(disks) GoogleSessions each with their own access tokens. If you can find out the reason why I'll undo these change and add a comment.

Reply:
I'm still looking, but the very original implementation had a borrowed compute_client:
https://github.com/hail-is/hail/pull/10090/files#diff-f8ba97f763395908a5b67f47a630c98e8d223ca5914f18d588f405d629d52197R958

Reply:
Found it! https://github.com/hail-is/hail/pull/10811

Reply:
Oh look it's me!

Reply:
Huh, I can't quite understand what I was getting at in that PR. Do you see any reason for prematurely closing the session credentials that that PR would fix?

Reply:
I think this was happening because we at the time (and maybe until this PR), had a lot of job cleanup that wasn't getting waited on by the task manager during worker shutdown. So the worker object closed the compute client that was still being used by Disks. I think the proper fix here is to make sure that the Worker properly waits for usages of the compute client to finish up before deleting it, instead of just using more clients.

Reply:
I'm pretty sure you are correct that this change will be fine, but any changes to the Disk handling code gives me anxiety because of the potential financial costs if we mess up and overlook something. Can we make sure the disk monitoring code with an alert is up to date in Grafana? At one point, I had a chart where we were alerted if there was more than X amount of storage in disks being used or X number of disks. I think it's $0.17 / GB / month, so we probably want to be alerted if we are spending more than $20 per hour on disk costs which is 87 TB (please double check the math). So we probably can do an alert on 25 TB or 1k disks as that will catch slower accumulations over time. What do you think?

Reply:
I'm all for making us feel confident in the change, so I'll do whatever metric we need to put in. Could you elaborate though on what could go wrong? I'd rather have an alert mean that something is wrong in our system and not that our scale is large.

Reply:
What went wrong before is we were not cleaning up disks and they were slowly accumulating until we were paying for exabytes of disks -- I think we had over 100k 10Gi disks and didn't realize it. I'm worried that if the compute client suddenly doesn't work because the client session is closed (or some unknown issue with this change), then we're going to slowly accumulate orphaned disks. There's a garbage collection loop on the driver that also checks for orphaned disks and deletes them so there is redundancy here. Maybe we put an alert on how many orphaned disks we are cleaning up?

Reply:
> Maybe we put an alert on how many orphaned disks we are cleaning up?

This sounds great, I'll set up this metric.